### PR TITLE
Check for existance of VM disk image and mark VM as DNB if image is missing.

### PR DESCRIPTION
**Pull request by activeshadow**

### CODE REVIEW

File: phenix/app/startup.go [hunk 1]
```diff
@@ -102,6 +102,9 @@ func (this *Startup) Configure(exp *types.Experiment) error {
 func (this Startup) PreStart(exp *types.Experiment) error {
 	// note in the mako file that there does not appear to be timezone or hostname for rhel and centos
 	startupDir := exp.Spec.BaseDir + "/startup"
+	
+	// currently assuming /phenix/images for image directory
+	imageDir := "/phenix/images/"
```
Comment (activeshadow) [inline phenix/app/startup.go:107]: @dnkcom for now this is fine here, but at some point soon we need to make this a accessible as a global phenix option.

File: phenix/app/startup.go [hunk 2]
```diff
@@ -113,6 +116,17 @@ func (this Startup) PreStart(exp *types.Experiment) error {
 			continue
 		}
 
+		// check if the disk image is present, if not set do not boot to true
+		if _, err := os.Stat(imageDir + node.Hardware.Drives[0].Image); os.IsNotExist(err) {
+			dnb := true
+			node.General.DoNotBoot = &dnb
+		}
```
Comment (activeshadow) [inline phenix/app/startup.go:123]: @dnkcom I would suggest moving this to the very beginning of the `for-loop` it's currently in so it also gets executed if the VM is a router.

File: phenix/app/startup.go [hunk 3]
```diff
@@ -113,6 +116,17 @@ func (this Startup) PreStart(exp *types.Experiment) error {
...
+
+		// if do not boot is true, skip it and continue
+		if *node.General.DoNotBoot {
+			continue
+		}
```
Comment (activeshadow) [inline phenix/app/startup.go:128]: @dnkcom I would suggest leaving this out. This way, injections still get created even if marked as DNB. You never know when a user might want to override the disk image and force boot the VM anyway.

File: phenix/tmpl/templates/minimega_script.tmpl [hunk 4]
```diff
@@ -19,29 +19,33 @@ ns add-host localhost
 {{- range .Topology.Nodes }}
 {{/* added space to place hostname above relevant block */}}
 ## VM: {{ .General.Hostname }} ##
-    {{- if .General.Snapshot -}}
-    {{ $firstDrive := index .Hardware.Drives 0 }}
+    {{- if (derefBool .General.DoNotBoot) }}
```
Comment (activeshadow) [inline phenix/tmpl/templates/minimega_script.tmpl:22]: We have to dereference pointers to booleans in template files given how the temlate `if statement` works on pointers.

File: phenix/tmpl/tmpl.go [hunk 5]
```diff
@@ -16,9 +16,12 @@ import (
 // template.
 func GenerateFromTemplate(name string, data interface{}, w io.Writer) error {
 	funcs := template.FuncMap{
-		"add": func(a, b int) int {
+		"addInt": func(a, b int) int {
 			return a + b
 		},
+		"derefBool": func(b *bool) bool {
+			return *b
+		},
```
Comment (activeshadow) [inline phenix/tmpl/tmpl.go:24]: This is a new function available inside template files to dereference boolean pointers to get the actual boolean values.

File: phenix/app/startup.go (hunk 1 above)
Comment (dnkcom) [reply phenix/app/startup.go:107]: Agreed; we could set this in a global environment value.

File: phenix/tmpl/templates/minimega_script.tmpl (hunk 4 above)
Comment (dnkcom) [reply phenix/tmpl/templates/minimega_script.tmpl:22]: I further edited the template to remove whitespace (`bbd2edf`)

**Comment by activeshadow** [comment]
Whoops! This got merged manually as commit 4a73f2db.

**Pull request closed without merging**